### DISCUSSION TITLE
Output color mismatch (Dark) while processing RAW image formats like .NEF, .CR3, etc

### DISCUSSION THREAD

**Discussion by guru248** [ImageMagick/ImageMagick#6202] [General]
I am trying to generate a preview Jpeg image for raw image formats using ImageMagick-7.1.0-Q16-HDRI. But I can see that the resulting Jpegs are much **darker** than the original image as viewed on Photoshop. I am using the following command:
`magick -define dng:use-camera-wb=true "D:\Data\NEF.NEF" -thumbnail"1400x1400" +antialias "D:\Data\NEFPreview.jpg"`

This is happening for a lot of **RAW formats** (ARW, CR2, CR3, CRW, DNG, NEF, NRW). I've checked a lot of previous discussions, but nothing seems to work for me. 
Not sure if I am missing something, **any help would be much appreciated**.

Attaching screenshot of source(left) and preview (right) file:
--Edit Removing Screenshot

**Comment by snibgo** [comment]
> ... the resulting Jpegs are much darker than the original image as viewed on Photoshop.

Either Photoshop is showing you the embedded JPEG, or Photoshop is doing some processing of the image.

You can extract the embedded JPEG with exiftool.

Or you can process the image with IM, for example:
```
magick -define dng:use-camera-wb=true nef.nef -auto-level -auto-gamma -thumbnail "1400x1400" out.tiff
```
Your `+antialias` does nothing.

**Reply by guru248** [reply]
Hi, thanks for your imput. I tried extracting the embedded JPEG using exiftool, and it matches with what is displayed in Photoshop.
`exiftool -b -JpgFromRaw "D:\Data\NEF.NEF" > "D:\Data\NEFextract.jpg"`

Is there anyway to get this result via ImageMagick?

I need to handle 1000s of images of different formats, so using auto-level & -auto-gamma is not an ideal solution as it could distort other images.

**Comment by snibgo** [comment]
>Is there anyway to get this result via ImageMagick?

In delegates.xml, insert this line:
```
  <delegate decode="jfr" command="cmd.exe /c exiftool -b -JpgFromRaw &quot;%i&quot; >&quot;%o&quot;"/>
```
That line must be somewhere within the `<delegatemap> ... </delegatemap>` lines.

[EDIT: That line is suitable for Windows, which you use. The line might be different for Unix etc.]

Then you can do this:
```
magick -verbose jfr:nef.nef x.tiff
```
[EDIT: You don't need `-verbose`.]

IM will call exiftool to extract the jpeg, and read the image into IM's memory. Then the IM command saves the image into a tiff file.

I decided to use the prefix "jfr". You can invent any prefix you want, but don't use one that has another meaning, such as "png" or "nef".

**Comment by guru248** [comment]
Actually, my goal is to avoid using any executable files and use Magick.net class in dotnet to perform the task in a program. So, can we extract the Jpeg using just imageMagick command line and avoid exif tool? Or get a similar output to that of exif result?

**Comment by snibgo** [comment]
IM has no other method for extracting embedded JPEGs from raw camera files.

**Comment by snibgo** [comment]
Update to my previous post: See https://imagemagick.org/script/defines.php , which includes:

> -define dng:read-thumbnail=true 	Read the embedded thumbnail and store it as a profile called 'dng:thumbnail'.

I haven't explored this.

**Comment by guru248** [comment]
This looks to be helpful. Thanks!
